src/crimson/osd/osd_operation.cc

```diff
@@ -94,6 +94,7 @@ void OSDOperationRegistry::put_historic(const ClientRequest& op)
     assert(fastest_historic_iter != std::end(historic_registry));
     const auto& fastest_historic_op =
       static_cast<const ClientRequest&>(*fastest_historic_iter);
+    historic_registry.erase(fastest_historic_iter);
```

Comment:
OK, the `last_of_recents` should be safe as `erase()`:

> Note: Invalidates the iterators (but not the references) to the erased element.

Reply:
> `fastest_historic_op's` refcount will be decreased more
> than once in func `put_historic`, leading to `fastest_historic_op` be deleted ahead.

why the refcount of `fastest_historic_iter` is decreased twice ?

Reply:
As `fastest_historic_iter` is remaining in `historic_registry`, if it is not released, at the next round, `put_historic` will decrease the refcount the second time.

Reply:
thank you for your analysis and explanation!